client/src/app/pages/applications/components/application-assessment-status/tests/application-assessment-status.test.tsx

```diff
@@ -20,6 +20,110 @@ describe("useAssessmentStatus", () => {
     server.resetHandlers();
   });
 
+  it("Updates hasApplicationAssessmentInProgress to false when assessment is marked as not required", async () => {
+    server.use(
+      rest.get("/hub/archetypes", (req, res, ctx) => {
+        return res(
+          ctx.json([
+            createMockArchetype({
+              id: 1,
+              name: "archetype1",
+              applications: [],
+              assessed: false,
+              assessments: [],
+            }),
+          ])
+        );
+      }),
+      rest.get("/hub/assessments", (req, res, ctx) => {
+        return res(
+          ctx.json([
+            createMockAssessment({
+              id: 1,
+              application: { id: 1, name: "app1" },
+              questionnaire: { id: 1, name: "questionnaire1" },
+              status: "started",
+              required: true,
+              sections: [],
+            }),
+            createMockAssessment({
+              id: 2,
+              application: { id: 1, name: "app1" },
+              questionnaire: { id: 2, name: "questionnaire2" },
+              status: "complete",
+              required: true,
+              sections: [],
+            }),
+          ])
+        );
+      })
+    );
+
+    const queryClient = new QueryClient({
+      defaultOptions: {
+        queries: {
+          retry: false,
+          cacheTime: 1000,
+        },
+      },
+    });
+
+    const { result, rerender } = renderHook(
+      () => useAssessmentStatus(createMockApplication({ id: 1, name: "app1" })),
+      { queryClient }
+    );
+
+    await waitFor(() => {
+      expect(result.current.hasApplicationAssessmentInProgress).toBe(true);
+    });
+
+    server.use(
+      rest.get("/hub/archetypes", (req, res, ctx) => {
+        return res(
+          ctx.json([
+            createMockArchetype({
+              id: 1,
+              name: "archetype1",
+              applications: [],
+              assessed: false,
+              assessments: [],
+            }),
+          ])
+        );
+      }),
+      rest.get("/hub/assessments", (req, res, ctx) => {
+        return res(
+          ctx.json([
+            createMockAssessment({
+              id: 1,
+              application: { id: 1, name: "app1" },
+              questionnaire: { id: 1, name: "questionnaire1" },
+              status: "started",
+              required: false,
+              sections: [],
+            }),
+            createMockAssessment({
+              id: 2,
+              application: { id: 1, name: "app1" },
+              questionnaire: { id: 2, name: "questionnaire2" },
+              status: "complete",
+              required: false,
+              sections: [],
+            }),
+          ])
+        );
+      })
```

Comment:
Having the second set of handlers is a bit confusing.

I suggest just doing a use once kind of setup so it is more obvious that the data will change on the second request.  

This may work (not tested):
```jsx
    rest.get("/hub/assessments", (req, res, ctx) => {
        return res.once(
          // first json
        );
      })
    rest.get("/hub/assessments", (req, res, ctx) => {
        return res(
          // json for the rest of the time
        );
      })
```

See: https://v1.mswjs.io/docs/api/setup-server/use#one-time-override

Reply:
Or I just found you can use a generator function in this case -- way cooler:
https://github.com/mswjs/msw/discussions/1117#discussioncomment-3124548

It points at this bit of code for more examples: https://github.com/mswjs/msw/blob/6990c3a57a1f515fe977c2eaa7741005b885b8af/test/rest-api/generator.mocks.ts#L4

//cc @rszwajko